packages/core/src/rules/common/no-duplicated-tag-names.ts

```diff
@@ -0,0 +1,27 @@
+import { getTagName } from '../utils.js';
+
+import type { Async2Rule, Async3Rule, Oas2Rule, Oas3Rule } from '../../visitors.js';
+import type { UserContext } from '../../walk.js';
+
+export const NoDuplicatedTagNames: Oas3Rule | Oas2Rule | Async2Rule | Async3Rule = ({
+  ignoreCase = false,
+}) => {
+  const tagNames = new Set<string>();
+
+  return {
+    Tag: {
+      leave(tag: { name: string }, ctx: UserContext) {
+        const tagName = getTagName(tag, ignoreCase);
+
+        if (tagNames.has(tagName)) {
+          ctx.report({
+            message: `Duplicate tag name found: '${tag.name}'`,
```

Comment:
One more thing: we try to follow sentence case in the error messages:

```suggestion
            message: `Duplicate tag name found: '${tag.name}'.`,
```

Could you update that everywhere when you have time?

Reply:
Thank you, i did it.

Reply:
Could you update snapshots accordingly as well?

Reply:
I see a failed unit test. I will investigate the issue, but I ran all tests yesterday locally before pushing and i run unit tests now, everything passed

Reply:
There were recent changes to the main branch that might cause the diffs in snapshots.